src/filesystem/unix/SDL_sysfilesystem.c

```diff
@@ -332,4 +333,122 @@ SDL_GetPrefPath(const char *org, const char *app)
     return retval;
 }
 
+char *
+getpath_try_xdg(SDL_Folder folder)
+{
+    const char *param = NULL;
+    char cmd[32]; /* 32 bytes should be enough, increase if needed */
+    FILE *xdg_proc;
+    char buffer[4096]; /* Apparently PATH_MAX is unreliable */
+    size_t bytes_read;
+    char *retval;
+
+    /* According to `man xdg-user-dir`, the possible values are:
+        DESKTOP
+        DOWNLOAD
+        TEMPLATES
+        PUBLICSHARE
+        DOCUMENTS
+        MUSIC
+        PICTURES
+        VIDEOS
+    */
+    switch(folder)
+    {
+        case SDL_FOLDER_HOME:
+            return SDL_getenv("HOME");
+
+        case SDL_FOLDER_DESKTOP:
+            param = "DESKTOP";
+            break;
+
+        case SDL_FOLDER_DOCUMENTS:
+            param = "DOCUMENTS";
+            break;
+
+        case SDL_FOLDER_DOWNLOADS:
+            param = "DOWNLOADS";
+            break;
+
+        case SDL_FOLDER_MUSIC:
+            param = "MUSIC";
+            break;
+
+        case SDL_FOLDER_PICTURES:
+            param = "PICTURES";
+            break;
+
+        case SDL_FOLDER_VIDEOS:
+            param = "VIDEOS";
+            break;
+
+        case SDL_FOLDER_SCREENSHOTS:
+        default:
+            SDL_SetError("No corresponding XDG user directory");
+            return NULL;
+    }
+
+    /* param *should* to be set to something at this point, but just in case */
+    if (!param)
+    {
+        SDL_SetError("No corresponding XDG user directory");
+        return NULL;
+    }
+
+    if (snprintf(cmd, sizeof(cmd), "xdg-user-dir %s", param) >= sizeof(cmd))
```

Comment:
It seems to me like it would be better to parse `user-dirs.dirs` directly, instead of calling out to a helper executable that isn't guaranteed to be installed on all systems (which is a relatively "expensive" thing to do - forking a subprocess does a lot more than reading a file, especially if you're a game process that is using a lot of memory).

For example, [GLib's implementation](https://gitlab.gnome.org/GNOME/glib/-/blob/2.76.2/glib/gutils.c?ref_type=tags#L2228) is adapted from [the implementation of the xdg-user-dir command](https://gitlab.freedesktop.org/xdg/xdg-user-dirs/-/blob/master/xdg-user-dir-lookup.c), which is permissively-licensed. One of the reasons I linked to other libraries' prior art in the issue was so that anyone interested in this feature can compare with their implementations, not just their APIs.

Reply:
Ah, I didn't see GLib's implementation was permissive! I only looked at the top, saw the LGPL boiler plate, and got too scared to read the code by fear of committing copyright infringement. Had I expressed that fear, I would've been told the relevant part is permissive. Sorry for my omission!